python/social_bot/envs/grocery_ground.py

```diff
@@ -0,0 +1,247 @@
+# Copyright (c) 2019 Horizon Robotics. All Rights Reserved.
+"""
+A simple enviroment for an agent play on a groceryground
+"""
+import os
+import logging
+import numpy as np
+import random
+import PIL
+
+import gym
+from gym import spaces
+from collections import OrderedDict
+
+import social_bot
+from social_bot import teacher
+from social_bot.teacher import TeacherAction
+from social_bot import goal_task
+from social_bot.goal_task import GoalTask
+import social_bot.pygazebo as gazebo
+
+logger = logging.getLogger(__name__)
+
+
+class GroceryGroundGoalTask(GoalTask):
+    """
+    For this task, the agent will receive reward 1 when it is close enough to the goal.
+    If it is moving away from the goal too much or still not close to the goal after max_steps,
+    it will get reward -1.
+    """
+
+    def __init__(self,
+                 max_steps=500,
+                 goal_name="bookshelf",
+                 success_distance_thresh=0.5,
+                 fail_distance_thresh=3.0,
+                 goal_random_range=10.0):
+        """
+        Args:
+            max_steps (int): episode will end if not reaching goal in so many steps
+            goal_name (string): name of the goal in the world
+            success_distance_thresh (float): the goal is reached if it's within this distance to the agent
+            fail_distance_thresh (float): if the agent moves away from the goal more than this distance,
+                it's considered a failure and is givne reward -1
+            goal_random_range (float): the goal's random position range
+            
+        """
+        self._goal_random_range = goal_random_range
+        super(GroceryGroundGoalTask, self).__init__(
+            max_steps=max_steps,
+            goal_name=goal_name,
+            success_distance_thresh=success_distance_thresh,
+            fail_distance_thresh=fail_distance_thresh)
+
+    def _move_goal(self, goal, agent_loc):
+        while True:
+            loc = (random.random() * self._goal_random_range -
+                   self._goal_random_range / 2,
+                   random.random() * self._goal_random_range -
+                   self._goal_random_range / 2, 0)
+            self._initial_dist = np.linalg.norm(loc - agent_loc)
+            if self._initial_dist > 0.5:
+                break
+        goal.set_pose((loc, (0, 0, 0)))
+
+
+class DiscreteSequence(gym.Space):
```

Comment:
This is same as the one in simple_navigation. We should not copy it.

Reply:
Move it to teacher class. Both simple_navigation and grocery_gournd import DiscreteSequence from teacher now.